src/wasm-linker.cpp

```diff
@@ -209,7 +209,13 @@ void Linker::layout() {
         param->type = type;
         call->operands.push_back(param);
       }
-      block->list.push_back(call);
+      Expression* e = call;
+      if (target->result != none) {
+        auto* drop = out.wasm.allocator.alloc<Drop>();
```

Comment:
I had it that way to match the rest of that function which didn't use a builder either. But I went ahead and made that whole section use a builder instead.